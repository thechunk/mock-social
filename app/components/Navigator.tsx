import React from 'react';
import {createStackNavigator} from 'react-navigation';
import UserList from "./UserList";
import {UsersApi} from "../classes/Api";

const Navigator = createStackNavigator({
    UserList: { screen: () => <UserList api={UsersApi} /> }
});
export default Navigator;
