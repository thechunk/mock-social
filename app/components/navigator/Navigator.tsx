import React from 'react';
import {
    createStackNavigator,
    NavigationInjectedProps,
    NavigationScreenConfig,
    NavigationScreenOptions
} from 'react-navigation';
import {UserList} from '../user-list/index';
import {UserDetail} from '../user-detail/index';
import {PhotoGrid} from '../photo-grid/index';
import {AlbumList} from '../album-list/index';
import {ImageViewer} from '../image-viewer/index';
import {BlurView} from '@react-native-community/blur';
import g from '../../styles/global';
import styles from './styles';
import {View} from "react-native";

const blurredHeaderOptions: NavigationScreenConfig<NavigationScreenOptions> = {
    headerBackground: (
        <View style={[g.flex1, styles.stackHeader]}>
            <BlurView
                blurType='xlight'
                style={g.flex1} />
        </View>
    ),
    headerTransparent: true
};

const Navigator = createStackNavigator({
    UserList: {
        screen: UserList,
        navigationOptions: { title: 'Users' }
    },
    UserDetail: {
        screen: UserDetail,
        navigationOptions: ({navigation}: NavigationInjectedProps) => ({
            title: navigation.getParam('hydrate').username,
            ...blurredHeaderOptions
        })
    },
    AlbumList: {
        screen: AlbumList,
        navigationOptions: { title: 'Albums' }
    },
    PhotoGrid: {
        screen: PhotoGrid,
        navigationOptions: ({navigation}: NavigationInjectedProps) => ({
            title: navigation.getParam('hydrate').title,
            ...blurredHeaderOptions
        })
    },
    ImageViewer: {
        screen: ImageViewer,
        navigationOptions: ({navigation}: NavigationInjectedProps) => ({
            title: navigation.getParam('hydrate').title,
            ...blurredHeaderOptions
        })
    }
});
export default Navigator;
