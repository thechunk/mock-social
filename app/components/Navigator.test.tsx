import * as React from 'react';
import Navigator from './Navigator';

import {shallow} from 'enzyme';

describe('Navigator', () => {
    it('renders correctly', () => {
        expect(shallow(<Navigator />)).toMatchSnapshot();
    });
});
