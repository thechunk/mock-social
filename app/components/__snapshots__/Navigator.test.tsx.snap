// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigator renders correctly 1`] = `
<Navigator
  onGestureBegin={[Function]}
  onGestureCanceled={[Function]}
  onGestureEnd={[Function]}
  onTransitionStart={[Function]}
/>
`;
